sympy/physics/mechanics/_geometry.py

```diff
@@ -175,13 +162,14 @@ def _point_is_on_surface(self, point: Point) -> bool:
         """
         point_vector = point.pos_from(self.point)
         if isinstance(point_vector, Vector):
-            point_radius = dot(point_vector, point_vector)
+            point_radius_squared = dot(point_vector, point_vector)
         else:
-            point_radius = point_vector**2
-        return Eq(point_radius, self.radius**2) == True
+            point_radius_squared = point_vector**2
+        return Eq(point_radius_squared, self.radius**2) == True
```

Comment:
I don't see how this is generally going to work. An `Eq()` call does not simplify an expression, so how with this work when the tested point has a very complex position expression?

I think it may be better to use "garbage in / garbage out" here and tell the user that these points must lie on the surface.

Reply:
My assumption in writing this was that the user would supply a (constant) symbol for the radius. If that's the case then this is valid and works. But if the user does supply a complex expression for the radius instead then I can see this breaking, even if the point is still on the surface.

Reply:
Just gave some warnings in the docstring.